src/sync/sync_user.hpp

```diff
@@ -117,24 +106,36 @@ struct SyncUserIdentity {
     std::string provider_type;
 
     SyncUserIdentity(const std::string& id, const std::string& provider_type);
+
+    bool operator==(const SyncUserIdentity& other) const
+    {
+        return id == other.id && provider_type == other.provider_type;
+    }
+
+    bool operator!=(const SyncUserIdentity& other) const
+    {
+        return id != other.id || provider_type != other.provider_type;
+    }
 };
 
 // A `SyncUser` represents a single user account. Each user manages the sessions that
 // are associated with it.
 class SyncUser {
 friend class SyncSession;
 public:
-    enum class State {
+    enum class State : std::size_t {
         LoggedOut,
+        LoggedIn,
```

Comment:
It looks like there's some duplication of state here; ie Active implies LoggedIn. Since we're already tracking the active user via `m_metadata_manager->set_current_user_identity` can we simplify this by removing the Active state entirely?

Reply:
Good call, that would be a nice bit of simplification.